recommendations/intro.adoc

```diff
@@ -652,8 +652,24 @@ WITH COUNT(m) AS numbermovies, SUM(x.rating * y.rating) AS xyDotProduct,
 SQRT(REDUCE(xDot = 0.0, a IN COLLECT(x.rating) | xDot + a^2)) AS xLength,
 SQRT(REDUCE(yDot = 0.0, b IN COLLECT(y.rating) | yDot + b^2)) AS yLength,
 p1, p2 WHERE numbermovies > 10
-RETURN p1.name, p2.name, xyDotProduct / (xLength * yLength) AS sim 
-ORDER BY sim DESC LIMIT 100;
+RETURN p1.name, p2.name, xyDotProduct / (xLength * yLength) AS sim
+ORDER BY sim DESC
+LIMIT 100;
+----
+
+We can also compute this measure using the https://neo4j.com/docs/graph-algorithms/current/algorithms/similarity-cosine[Cosine Similarity algorithm^] in the Neo4j Graph Algorithms Library.
+
+.Find the users with the most similar preferences to Cynthia Freeman, according to cosine similarity function
+[source, cypher]
+----
+MATCH (p1:User {name: 'Cynthia Freeman'})-[x:RATED]->(movie)
+MATCH (p2:User)-[x2:RATED]->(movie) WHERE p2 <> p1
+WITH p1, p2, collect(x.rating) AS p1Ratings, collect(x2.rating) AS p2Ratings
```

Comment:
should we collect the x.rating earlier to not have them explode according by the 2nd match?

Reply:
we can't - they need to refer to the same `movie` node. I'll move it onto one line so it's more clear what the query does.